sunpy/timeseries/sources/goes.py

```diff
@@ -173,8 +174,10 @@ def _parse_hdus(cls, hdulist):
         else:
             raise ValueError("Don't know how to parse this file")
 
-        times = [start_time + datetime.timedelta(seconds=int(np.floor(s)),
-                                                    microseconds=int((s - np.floor(s)) * 1e6)) for s in seconds_from_start]
+        times = [start_time + TimeDelta(int(np.floor(s))*u.second +
+                                        int((s - np.floor(s)) * 1e6))*u.microsecond
+                                        for s in seconds_from_start]
+                                        # Why not just pass s???
```

Comment:
Is the result the same if you just pass in `seconds_from_start` instead of having the `for` statement?